docs/application/web/guides/machine-learning/pipeline.md

```diff
@@ -0,0 +1,445 @@
+
+       return 0;
+   };
+
+   tizen.ml.pipeline.registerCustomFilter('flattenFilter', flattenFilter, inputInfo, outputInfo);
+   ```
+
+   > [!Note]
+   > For performance reasons `inputData` and `outputData` passed to `flattenFilter` behave differently than ordinary `TensorsData` objects.
+   > They cannot be disposed and are invalidated when the callback returns - you have to copy them manually to use outside the callback.
+   > `inputData` is read-only and `outputData` is initialized with random values.
+
+3. Create a pipeline with a `custom-easy-filter` element:
+   ```javascript
+   var pipelineDescription = 'videotestsrc num-buffers=3 ' +
+                             '! video/x-raw,width=20,height=15,format=BGRA ' +
+                             '! tensor_converter ' +
+                             '! tensor_filter framework=custom-easy model=flattenFilter ' +
+                             '! fakesink';
+   var pipeline = tizen.ml.pipeline.createPipeline(pipelineDescription);
+   ```
+
+When you will start the pipeline, the `flattenFilter` will transform 3-dimensional tensors into 1-dimensional vectors.
+
+
+## Use saved models
+Using machine learning models trained with popular frameworks, like [TensorFlow](https://tensorow.org), is one of the main use cases for NNStreamer pipelines.
+
+To read a model from file and use it in a pipeline:
+1. You have to know the absolute path to the saved model file. It can be obtained from a path relative to a virtual root:
+   ```javascript
+   var modelPath = 'documents/mobilenet_v1_1.0_224_quant.tflite';
+   var URI_PREFIX = 'file://';
+
+   var absoluteModelPath = tizen.filesystem.toURI(modelPath).substr(URI_PREFIX.length);
+   ```
+
+2. Create a pipeline with a `tensor_filter` node, with `model` property set to the location of a model.
+   ```javascript
+   var pipelineDescription = 'appsrc name=srcx  ' +
+                          '! other/tensor,dimension=(string)224:224:3:1,type=(string)float ' +
+                          '! tensor_filter framework=tflite model=' + absoluteModelPath + ' ' +
+                          '! appsink name=sinkx';
+   var pipeline = tizen.ml.pipeline.createPipeline(pipelineDescription);
+   ```
+
+3. Set up other pipeline elements. We will use a custom `Source` and `Sink`:
+   ```javascript
+   var source = pipeline.getSource('srcx');
+
+   function sinkListener(sinkName, data) {
+     var rawData = data.getTensorRawData(0);
+     console.log(sinkName + ' received the data:');
+     console.log(rawData);
+   };
+
+   pipeline.registerSinkListener('sinkx', sinkListener);
+   ```
+
+4. Run the pipeline. We will use random input:
+   ```javascript
+   pipeline.start();
+
+   var inputTensorsData = inputTensorsInfo.getTensorsData(0);
+   var randomInput = [];
+   for (var i = 0; i < 224 * 224 * 3; i++) {
+       randomInput.push(Math.random());
+   };
+   inputTensorsData.setTensorRawData(0, randomInput);
+   source.inputData(inputTensorsData);
+   ```
```

Comment:
## Use saved models

Using machine learning models trained with popular frameworks like [TensorFlow](https://tensorow.org) is one of the main use cases for NNStreamer pipelines.

To read a model from file and use it in a pipeline:

1. You have to know the absolute path to the saved model file. It can be obtained from a path relative to a virtual root:

   ```javascript
   var modelPath = 'documents/mobilenet_v1_1.0_224_quant.tflite';
   var URI_PREFIX = 'file://';
   var absoluteModelPath = tizen.filesystem.toURI(modelPath).substr(URI_PREFIX.length);
   ```
2. Create a pipeline with a `tensor_filter` node with `model` property set to the location of a model.
   
    ```javascript
       var pipelineDescription = 'appsrc name=srcx  ' +
                          '! other/tensor,dimension=(string)224:224:3:1,type=(string)float ' +
                          '! tensor_filter framework=tflite model=' + absoluteModelPath + ' ' +
                          '! appsink name=sinkx';
       var pipeline = tizen.ml.pipeline.createPipeline(pipelineDescription);
       ```
3. Set up other pipeline elements. We will use a custom `Source` and `Sink`:

    ```javascript
       var source = pipeline.getSource('srcx');
       function sinkListener(sinkName, data) {
       var rawData = data.getTensorRawData(0);
       console.log(sinkName + ' received the data:');
       console.log(rawData);
   };
   pipeline.registerSinkListener('sinkx', sinkListener);
   ```

4. Run the pipeline using  a random input:
  
     ```javascript
       pipeline.start();
       var inputTensorsData = inputTensorsInfo.getTensorsData(0);
       var randomInput = [];
       for (var i = 0; i < 224 * 224 * 3; i++) {
       randomInput.push(Math.random());
   };
   inputTensorsData.setTensorRawData(0, randomInput);
   source.inputData(inputTensorsData);
   ```suggestion